### PR TITLE
feat(storage) adds User Agent to Storage requests

### PR DESCRIPTION
**Pull request by brennanMKE**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageMultipartUploadClient.swift [hunk 1]
```diff
@@ -100,11 +100,7 @@ class DefaultStorageMultipartUploadClient: StorageMultipartUploadClient {
             var request = URLRequest(url: preSignedURL)
             request.cachePolicy = .reloadIgnoringLocalCacheData
             request.httpMethod = "PUT"
-
-            /*
-            let userAgent = AWSServiceConfiguration.baseUserAgent().appending(" MultiPart")
-            request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
-             */
+            request.setValue(serviceProxy.userAgent, forHTTPHeaderField: "User-Agent")
```
Comment (ruisebas) [inline AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageMultipartUploadClient.swift:103]: We no longer need the `"MultiPart"` suffix?

File: AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift [hunk 2]
```diff
@@ -104,6 +105,8 @@ class AWSS3StorageService: AWSS3StorageServiceBehaviour, StorageServiceProxy {
         self.awsS3 = awsS3
         self.bucket = bucket
 
+        self.userAgent = "\(AmplifyAWSServiceConfiguration.frameworkMetaData())"
```
Comment (ruisebas) [inline AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift:108]: This does not seem to have the same data and format as V1's `AmplifyAWSServiceConfiguration.baseUserAgent()`. 

Is that OK/expected?

v1: `amplify-iOS/1.28.1 iOS/16.0 en_US`
v2: `lib/amplify-ios/1.27.1-swift-sdk-dev-preview.0`

File: AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageMultipartUploadClient.swift (hunk 1 above)
Comment (brennanMKE) [reply AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Support/Internal/StorageMultipartUploadClient.swift:103]: We will need guidance on how metrics are tracked.

File: AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift (hunk 2 above)
Comment (royjit) [reply AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift:108]: Here is the useragent string from Swift SDK:

```aws-sdk-swift/1.0 api/cognito-identity-provider/1.0 os/macOS/12.6.0 lang/swift/5.7 lib/amplify-ios/1.27.1-swift-sdk-dev-preview.0, ```

**Comment by jcjimenez** [comment]
More recent revision: https://github.com/aws-amplify/amplify-swift/pull/2954

**Pull request closed without merging**